### PR TITLE
Ghost on App Engine update

### PR DESCRIPTION
**Pull request by hnipps**
I've updated the tutorial for deploying Ghost on Google Cloud App Engine. I tried to follow the tutorial myself, but noticed that Ghost have changed some things, notably the config files. So I updated this tutorial with the process I used to deploy.
Major changes:
- Updated example config files
- Added an extra step that involves running knex-migrator with the 'production' flag set

### CODE REVIEW

**Comment by hnipps** [comment]
@jmdobry I kept your name in here as an author, because you created the original article. As @googlebot says, please can you confirm that you're okay with your commits being contributed to his project?

File: tutorials/ghost-on-app-engine-part-1-deploying.md [hunk 1]
```diff
@@ -293,9 +281,41 @@ Create a `config.js` file from the default config file:
...
+            "server": {
+                "host": "0.0.0.0",
+                "port": "8080"
+            },
+            "paths": {
+                "contentPath": "content/"
+            }
+        }
+
+It's very important that you only do this step after migrating the database, because the socketPath property is required to deploy on Google App Engine, but it causes knex-migrator to throw an error.
```
Comment (parkjam4) [inline tutorials/ghost-on-app-engine-part-1-deploying.md]: I would split this into 2 sentences:
It's very important that you only do this step after migrating the database. The *socketPath* property is required to deploy on Google App Engine, but it causes *knex-migrator* to throw an error.

File: tutorials/iot-device-to-device/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# Device to Device demo
+
+Demonstrates how to relay messages from one device to another device using
+Google Cloud functions.
```
Comment (parkjam4) [inline tutorials/iot-device-to-device/README.md]: Capitalize Functions.

File: tutorials/iot-device-to-device/README.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+From the [Google Cloud Console](https://console.cloud.google.com/iot) create
+a Google Cloud IoT Core Device Registry. When you create it, either set the
+event notification topic to an existing PubSub topic or create a new topic to
+be used for the demo.
+
+### Upload the Google Cloud Function
+
+[The dev2dev instructions](dev2dev/README.md) explain how to upload the
+Google Cloud function to relay messages from one device to another.
```
Comment (parkjam4) [inline tutorials/iot-device-to-device/README.md]: Capitalize Function.

File: tutorials/iot-device-to-device/index.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
+---
+title: Google Cloud IoT Core Device to Device Communication
```
Comment (parkjam4) [inline tutorials/iot-device-to-device/index.md]: Perhaps start the title with an action, such as Configure Google Cloud IoT Core Device to Device Communication

File: tutorials/iot-device-to-device/index.md [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+description: Learn how to relay messages from one device to another using Google Cloud Functions.
+author: gguuss
+tags: Google Cloud IoT Core, Cloud IoT, Google Cloud Functions
+date_published: 2017-12-05
+---
+
+Sometimes you want to have one device control another device. For example, lets
```
Comment (parkjam4) [inline tutorials/iot-device-to-device/index.md]: change lets to let's

File: tutorials/iot-device-to-device/index.md [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+      client = authClient.createScoped(scopes);
+    }
+
+    google.options({auth: authClient});
+    //...
+  }
+```
+
+The Google Cloud Function exported by the script is titled `relayCloudIot` and
+is specifed as:
```
Comment (parkjam4) [inline tutorials/iot-device-to-device/index.md]: change specifed to specified

File: tutorials/iot-device-to-device/index.md [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+      console.log('message received: ', Buffer.from(message, 'base64').toString('ascii'));
+      let payload = JSON.parse(Buffer.from(message, 'base64').toString('ascii'));
+      console.log(`${payload.hops} to ${++payload.hops}`);
+      publishAsync(payload.hops, payload.hops+1);
+    });
+```
+
+To run the demo after you have successfully setup your Device Registry and
+deployed your Google Cloud Function, run the following command from the
+virtualdevice folder:
```
Comment (parkjam4) [inline tutorials/iot-device-to-device/index.md]: change virtualdevice to *virtualdevice*

File: tutorials/iot-device-to-device/virtualdevice/README.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
+<img src="https://avatars2.githubusercontent.com/u/2810941?v=3&s=96" alt="Google Cloud Platform logo" title="Google Cloud Platform" align="right" height="96" width="96"/>
+
+# Google Cloud IoT Core Virtual Device demo
+
+This sample app publishes data to Cloud Pub/Sub using the MQTT bridge provided
+as part of Google Cloud IoT Core.
+
+Note that before you can run this sample, you must register a device as
+described in the parent README.
```
Comment (parkjam4) [inline tutorials/iot-device-to-device/virtualdevice/README.md]: possible to link to the README file?

File: tutorials/iot-device-to-device/virtualdevice/README.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+# Running the sample
+
+The following command summarizes the sample usage:
+
+  Usage: virtualdevice [options]
+
+  Example Google Cloud IoT Core MQTT device connection code.
+
+  Options:
```
Comment (parkjam4) [inline tutorials/iot-device-to-device/virtualdevice/README.md]: Need to format this table.

File: tutorials/ghost-on-app-engine-part-1-deploying.md [hunk 10]
```diff
@@ -1,7 +1,7 @@
 ---
-title: Ghost on App Engine Flexibe Environment Part 1 - Deploying
+title: Ghost on App Engine Flexible Environment Part 1 - Deploying
```
Comment (parkjam4) [inline tutorials/ghost-on-app-engine-part-1-deploying.md]: Would be nice to change this title to Deploying Ghost on App Engine Flexible Environment - Part 1
If you did this, then you would need to make a similar change to Part 2

File: tutorials/iot-device-to-device/README.md [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+        --projectId<your-project-id> \
+        --deviceId=<your-device-id>\
+        --privateKeyFile=../rsa_private.pem \
+        --algorithm=RS256 \
+        --numMessages=1 \
+        --registryId=<your-registry-id>
+```
+
+When the sample runs, the virtual device will trigger configuration updates
+from the Google Cloud Functions.
```
Comment (parkjam4) [inline tutorials/iot-device-to-device/README.md]: We typically do not use articles in front of the product names. So, no "the Google Cloud Functions."

File: tutorials/iot-device-to-device/README.md [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+
+## Troubleshooting
+
+When you run the sample code, run the following command to print the output
```
Comment (parkjam4) [inline tutorials/iot-device-to-device/README.md]: We typically do not use articles in front of the product names. So, no the Google Cloud Functions.

File: tutorials/iot-device-to-device/README.md [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+from the Google Cloud Function:
+
+```
+    gcloud beta functions logs read
+```
+
+This is useful for debugging aspects of the demo that run on the Google Cloud
```
Comment (parkjam4) [inline tutorials/iot-device-to-device/README.md]: This use of the article is ok though since you are referring to a callback for Google Cloud Functions

**Comment by hnipps** [comment]
I think I made a mistake with the branch I'm trying to merge into. I should be merging into "master" not "patch-1". Sorry for wasting people's time!
@parkjam4 I'll make the changes you suggested before creating a PR into "master".

**Pull request closed without merging**